components/proxy/src/main/java/com/hotels/styx/admin/AdminServerBuilder.java

```diff
@@ -82,68 +86,104 @@
-        httpRouter.add("/admin/tasks/plugin/", new PluginToggleHandler(plugins));
-
-        // Plugins Handler
-        routesForPlugins().forEach(route -> httpRouter.add(route.path(), route.handler()));
-
-        httpRouter.add("/admin/plugins", new PluginListHandler(plugins));
-
-        return new NettyServerBuilderSpec("Admin", environment.serverEnvironment(), new WebServerConnectorFactory())
+        Iterable<IndexHandler.Link> links = indexLinkPaths();
+
+        RouterBuilder httpRouter = new RouterBuilder()
+                .add("/", new IndexHandler(links))
+                .add("/version.txt", new VersionTextHandler(styxConfig.versionFiles()))
+                .add("/admin", new IndexHandler(links))
+                .add("/admin/status", new StatusHandler(healthCheckHandler))
+                .add("/admin/ping", new PingHandler())
+                .add("/admin/threads", new ThreadsHandler())
+                .add("/admin/metrics", new MetricsHandler(environment.metricRegistry(), metricsCacheExpiration))
+                .add("/admin/healthcheck", healthCheckHandler)
+                .add("/admin/configuration", new StyxConfigurationHandler(staticConfiguration()))
+                .add("/admin/configuration/origins", new OriginsHandler(backendServicesRegistry))
+                .add("/admin/jvm", new JVMMetricsHandler(environment.metricRegistry(), metricsCacheExpiration))
+                .add("/admin/origins/status", new OriginsInventoryHandler(environment.eventBus()))
+                .add("/admin/configuration/logging", new LoggingConfigurationHandler(styxConfig.startupConfig().logConfigLocation()))
+                .add("/admin/configuration/startup", new StartupConfigHandler(styxConfig.startupConfig()))
+                .add("/admin/styx/proxy/status", new ProxyStatusHandler(environment.configStore()))
+
+                // Dashboard
+                .add("/admin/dashboard/data.json", dashboardDataHandler(styxConfig))
+                .add("/admin/dashboard/", new ClassPathResourceHandler("/admin/dashboard/"))
+
+                // Tasks
+                .add("/admin/tasks/origins/reload", new HttpMethodFilteringHandler(POST, new OriginsReloadCommandHandler(backendServicesRegistry)))
+                .add("/admin/tasks/origins", new HttpMethodFilteringHandler(POST, new OriginsCommandHandler(environment.eventBus())))
+                ;
+
+        environment.configStore().watch("plugins", List.class)
+                .map(list -> (List<NamedPlugin>) list)
+                .subscribe(plugins -> {
+                    routesForPlugins(plugins).forEach(route -> httpRouter.add(route.path(), route.handler()));
+                    httpRouter.add("/admin/tasks/plugin/", new PluginToggleHandler(plugins));
+                    httpRouter.add("/admin/plugins", new PluginListHandler(plugins));
+                });
```

Comment:
We don't necessarily need a watch functionality for this use case. You can just render the admin links every time a request comes in. This way you'll just query the config store and add the links to the end of the page.

Reply:
This code does not render links, it builds the HttpRouter.